### PR TITLE
Autosave .git folder (PS-1994)

### PR DESCRIPTION
**Pull request by JakubMatejka**
I refactored the file a little to be better comprehensible and testable. What do you think?

### CODE REVIEW

**Comment by JakubMatejka** [comment]
But what about the restore? We take all files with tag `sandbox-{id}` (https://github.com/keboola/sandboxes/blob/b66000ba3d0534181581257b906285d5429f855c/src/App.php#L412-L425) so we will have to check the file name and if it has `.tar.gz` suffix then ungzip. Or should we rather assign it some special tag? And the ungzipping itself should be done in data loader, what do you think?

File: notebookUtils.py [hunk 1]
```diff
@@ -27,13 +36,15 @@ def retrySession(
     return session
 
 
-def saveFile(file_path, token, log):
+def saveFile(file_path, sandbox_id, token, log):
```
Comment (pivnicek) [inline notebookUtils.py:39]: tohle imho dobry 👍

File: tests/notebookUtils.py [hunk 2]
```diff
@@ -1,46 +1,108 @@
...
+            assert 'file' in dataLoaderMock.last_request.text
+            assert 'tags' in dataLoaderMock.last_request.text
+
+            assert apiMock.call_count == 1
+            assert 'lastAutosaveTimestamp' in apiMock.last_request.text
+
+    def test_getStorageTokenFromEnvMissing(self):
+        os.environ.pop('KBC_TOKEN')
+        with pytest.raises(Exception):
+            getStorageTokenFromEnv(logging)
```
Comment (pivnicek) [inline tests/notebookUtils.py:58]: nice, new to me 👍

**Review by pivnicek — APPROVED** [review]
awesome!

I double checked that we're using a specific tag of this lib so we don't need to be afraid of that :-)

**Comment by JakubMatejka** [comment]
Added `git` tag to the backup: https://github.com/keboola/sandboxes-notebook-utils/pull/9/commits/c58790dcfa28e04214b960e10687cede3d1280f6

**Review by pivnicek — APPROVED** [review]
nice!

**Pull request merged**